nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/queue/PutAzureQueueStorage_v12.java

```diff
@@ -0,0 +1,160 @@
+        final List<ValidationResult> results = (List<ValidationResult>) super.customValidate(validationContext);
+        final int visibilityTimeout = validationContext.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+
+        if (visibilityTimeout <= 0) {
+            results.add(new ValidationResult.Builder()
+                    .valid(false)
+                    .subject(VISIBILITY_TIMEOUT.getDisplayName())
+                    .explanation(VISIBILITY_TIMEOUT.getDisplayName() + " should be greater than 0 secs")
+                    .build());
+        }
+
+        // 7 days is the maximum timeout as per https://learn.microsoft.com/en-us/rest/api/storageservices/put-message
+        final int maxVisibilityTimeout = 7 * 24 * 60 * 60;
+        if (visibilityTimeout >  maxVisibilityTimeout) {
+            results.add(new ValidationResult.Builder()
+                    .valid(false)
+                    .subject(VISIBILITY_TIMEOUT.getDisplayName())
+                    .explanation(VISIBILITY_TIMEOUT.getDisplayName() + " should not be greater than 7 days")
+                    .build());
+        }
+
+        final int ttl = validationContext.getProperty(TTL).asTimePeriod(TimeUnit.SECONDS).intValue();
+        if (ttl <= 0) {
+            results.add(new ValidationResult.Builder()
+                    .subject(TTL.getDisplayName())
+                    .valid(false)
+                    .explanation(TTL.getDisplayName() + " should be any positive number")
+                    .build());
+        }
+
+        return results;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final long startNanos = System.nanoTime();
+
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        session.exportTo(flowFile, baos);
+        final String flowFileContent = baos.toString();
+
+        final int visibilityTimeoutInSecs = context.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int ttl = context.getProperty(TTL).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int requestTimeoutInSecs = context.getProperty(REQUEST_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+
+        final QueueClient queueClient;
+        queueClient = createQueueClient(context, flowFile);
+        try {
+            queueClient.sendMessageWithResponse(
+                    flowFileContent,
+                    Duration.ofSeconds(visibilityTimeoutInSecs),
+                    Duration.ofSeconds(ttl),
+                    Duration.ofSeconds(requestTimeoutInSecs),
+                    Context.NONE
+            );
+        } catch (final QueueStorageException e) {
+            getLogger().error("Failed to write the message to Azure Queue Storage due to {}", new Object[]{e});
```

Comment:
The wrapping `new Object[]` is not necessary. The exception does not need to be parameterized in the log.
```suggestion
            getLogger().error("Failed to write message to Azure Queue Storage", e);
```

Reply:
Making the changes